pkg/controller/support/secrets/secrets_test.go

```diff
@@ -0,0 +1,264 @@
+				},
+			},
+			Want: map[string][]byte{"domino-test/foo/bar": []byte("goodbye")},
+		},
+		"errors for missing secrets": {
+			RequestedSecrets: []hephv1.SecretReference{{Namespace: "foo", Name: "bar"}},
+			ClientResponse:   []runtime.Object{},
+			WantError:        true,
+		},
+		"requires secrets to have hephaestus-accessible label": {
+			RequestedSecrets: []hephv1.SecretReference{{Namespace: "foo", Name: "bar"}},
+			ClientResponse: []runtime.Object{&corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "foo",
+					Name:      "bar",
+				},
+				Data: map[string][]byte{"baz": []byte("hello")},
+			}},
+			WantError: true,
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
+			// Assume a static domino-compute namespace for all ImageBuild requests
+			img := &hephv1.ImageBuild{
+				Status: hephv1.ImageBuildStatus{
+					Phase: hephv1.PhaseInitializing,
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "image-build-request",
+					Namespace: "domino-compute",
+				},
+				Spec: hephv1.ImageBuildSpec{
+					Secrets: tc.RequestedSecrets,
+				},
+			}
+
+			clientsetFunc = func(*rest.Config) (kubernetes.Interface, error) {
+				return fake.NewSimpleClientset(tc.ClientResponse...), nil
+			}
+
+			secretData, err := ReadSecrets(context.Background(), img, logr.Discard(), nil, nil)
+
+			if tc.WantError {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+				assert.Equal(t, tc.Want, secretData)
+			}
+		})
+	}
+}
+
+func TestReadSecretsTakesOwnership(t *testing.T) {
+	for name, tc := range map[string]struct {
+		RequestedSecret     hephv1.SecretReference
+		ReturnedSecret      *corev1.Secret
+		Want                map[string][]byte
+		WantImageBuildOwner bool
+	}{
+		"does not change owner by default": {
+			RequestedSecret: hephv1.SecretReference{Namespace: "domino-compute", Name: "foo"},
+			ReturnedSecret: &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "domino-compute",
+					Name:      "foo",
+					Labels:    map[string]string{"hephaestus-accessible": "true"},
+				},
+				Data: map[string][]byte{"bar": []byte("hello")},
+			},
+			Want: map[string][]byte{"domino-compute/foo/bar": []byte("hello")},
+		},
+		"updates owner reference to ImageBuild when hephaestus-owned label set": {
+			RequestedSecret: hephv1.SecretReference{Namespace: "domino-compute", Name: "foo"},
+			ReturnedSecret: &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "domino-compute",
+					Name:      "foo",
+					Labels: map[string]string{
+						"hephaestus-owned":      "true",
+						"hephaestus-accessible": "true",
+					},
+				},
+				Data: map[string][]byte{"bar": []byte("hello")},
+			},
+			Want:                map[string][]byte{"domino-compute/foo/bar": []byte("hello")},
+			WantImageBuildOwner: true,
+		},
+		"does not update owner references across namespaces, but still returns data": {
+			RequestedSecret: hephv1.SecretReference{Namespace: "domino-compute", Name: "foo"},
```

Comment:
is this test supposed to be testing a secret in a different namespace from the image build? seems like the image build is in `domino-compute`? (and if so, then, how is this test passing, since it seems like `WantImageBuildOwner` should be `true` for this setup?)